src/Policies.cpp

```diff
@@ -336,8 +355,13 @@ void PolicyManager::remove_policy(const std::string &name)
         _registry->handler_manager()->module_remove(name);
     }
 
-    if (!policy->input_stream()->policies_count()) {
+    if (policy->input_stream()->policies_count() <= 1) {
```

Comment:
let's not make any assumptions here - if there's only 1 policy left, get it and make sure it's the input_resources policy before trying to clean it up. we may make adding the input_resources policy optional, for example

Reply:
I've done this validation in another way, please check it